core/src/main/java/io/grpc/internal/InternalHandlerRegistry.java

```diff
@@ -31,28 +31,44 @@
 
 package io.grpc.internal;
 
+import io.grpc.ExperimentalApi;
 import io.grpc.ServerMethodDefinition;
 import io.grpc.ServerServiceDefinition;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import javax.annotation.Nullable;
 
 final class InternalHandlerRegistry {
+
+  private final List<ServerServiceDefinition> services;
   private final Map<String, ServerMethodDefinition<?, ?>> methods;
 
-  private InternalHandlerRegistry(Map<String, ServerMethodDefinition<?, ?>> methods) {
+  private InternalHandlerRegistry(List<ServerServiceDefinition> services,
+      Map<String, ServerMethodDefinition<?, ?>> methods) {
```

Comment:
break line at highest semantic level.     In this case, keep all arguments on the same line.